Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -129,21 +142,9 @@ public class ConnectionPool {
         if item.isConnected == false {
             releaser(item)
             capacity -= 1
-            if let newItem = generator() {
-                capacity += 1
-                pool.append(newItem)
-                semaphore.signal()
-            }
-        }
-        // If we took the last item, we can choose to grow the pool
-        if (pool.count == 0 && capacity < limit) {
-            if let newItem = generator() {
-                capacity += 1
-                pool.append(newItem)
-                semaphore.signal()
-            }
+            return populateConnectionPool()
```

Comment:
This isn't right - we should be returning an item that is _not_ (currently) in the pool.  It'll be returned to the pool later in `give()`.